_posts/2019-10-29-dhis2-and-docker.md

```diff
@@ -0,0 +1,214 @@
+		  volumes:  
+		    - ./config/dhis2_home/dhis.conf:/DHIS2_home/dhis.conf
+		  environment:
+		    - WAIT_FOR_DB_CONTAINER=db:5432 -t 0
+	      ports:  
+		    - "8080:8080"
+		  depends_on: 
+		    - db
+
+You probably noticed, that most of the file is the same as in the previous step, but we added _volumes_ and _environment_ properties for _db_ and _web_ containers. 
+
+Attaching the SQL file to the _db_ container will tell postgres to pre-populate db before starting the container. Depending on the file size, postgres might take a while to do this and we might run into race conditions when DHIS2 starts earlier than postgres and fails to connect. To solve this, we tell the _web_  container that it has to wait for postgres to finish before starting. We do this by using the WAIT_FOR_DB_CONTAINER environment variable.  
+You will notice the -t  option. The -t option stands for  _timeout_. 0  means that _web_ container will wait as long as it takes for _db_ container to become ready, but you can set any timeout (in seconds) you want.  
+
+Note, that you still need to configure connection properties in dhis.conf: 
+
+    connection.dialect = org.hibernate.dialect.PostgreSQLDialect  
+	connection.driver_class = org.postgresql.Driver  
+	
+	# "db" maps to service name defined in docker-compose
+	# "dhis2" maps to POSTGRES_DB defined in docker-compose
+	connection.url = jdbc:postgresql://db/dhis2 
+	
+	# maps to POSTGRES_USER environment variable in docker-compose.
+	connection.username = dhis 
+	
+	# maps to POSTGRES_PASSWORD environment variable in docker-compose.
+	connection.password = dhis 
+
+
+To run this file with docker compose, execute the following command: 
+`docker-compose -f /pathToDockerComposeFile up`
+
+or, if the file is on your current directory: 
+`docker-compose up`
+
+**Outcome**: docker-compose will start 2 containers: _db_ and _web_ and when tomcat fully starts up, you will be able to reach DHIS2 at http://localhost:8080. The database will be pre-populated. 
+
+To destroy the instance, run `docker-compose down`. 
+
+
+## Q&A 
+Q:  **How can I use your docker images in production?** 
+A: We can't recommend using docker in production yet. We know that there are implementations that runs DHIS2 in docker successfully, but we can't recommend doing so just yet. If you decide to go for it, make sure you perform enough security, performance and stress tests. 
+
+Q: **How can I contribute?** 
+A: Great question! We value all contributions and we want to have as many as possible! Firstly, you can contribute by testing our images and raising feature requests as well as bugs. You have ideas on how to make images more secure, performant or functional? We want to hear it! If you are using your own docker images, give our images a try and tell us how can we improve. We also accept code contributions. Docker stuff is located in [dhis2-core](https://github.com/dhis2/dhis2-core) git repository. 
+
+
+
+
+## Resources: 
```

Comment:
We can provide links to our docker-compose repo on GitHub as reference for the curious: https://github.com/dhis2/docker-compose

Reply:
oh wow didn't even know we had it